src/client/Client.cc

```diff
@@ -16060,18 +16058,24 @@ int64_t Client::ll_preadv_pwritev(struct Fh *fh, const struct iovec *iov,
       return retval;
     }
 
+    retval = 0;
+    std::unique_lock cl(client_lock);
```

Comment:
do we really need unique_lock?

Reply:
Using a unique_lock is the preferred way to achieve such a thing -- unlocking and locking it back again.

Reply:
ah, got it.